types/datum.go

```diff
@@ -1077,6 +1077,12 @@ func (d *Datum) convertToMysqlDecimal(sc *stmtctx.StatementContext, target *Fiel
 	if err == nil && err1 != nil {
 		err = err1
 	}
+	if dec.negative && mysql.HasUnsignedFlag(target.Flag) {
+		dec = &zeroMyDecimal
```

Comment:
This will reference `zeroMyDecimal` directly, then we may modify it accidentally.
use `*dec = zeroMyDecimal` instead.

Reply:
Done~